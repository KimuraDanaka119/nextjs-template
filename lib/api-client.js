import fetch from 'isomorphic-unfetch';
import config from 'configs/config';

export default class ApiClient {
  async doFetch(path) {
    const url = config.api.baseURL + path;

    const res = await fetch(url);
    const data = await res.json();

    return {
      statusCode: res.status,
      data,
    };
  }

  async getShows(keyword) {
    const res = await this.doFetch(`/search/shows?q=${keyword}`);

    return res;
  }

  async getShow(id) {
    const res = await this.doFetch(`/shows/${id}`);

    return res;
  }
}
