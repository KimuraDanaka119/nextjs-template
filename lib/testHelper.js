import ApiClient from './api-client';

function t(key) {
  return key;
}

const config = {
  env: 'test',
  versionHash: 'test',
  paths: {
    static: '/static/test',
    styles: '/static/test/styles',
    images: '/static/test/images',
    locales: '/static/test/locales',
  },
  api: {
    baseURL: '.',
  },
};

const apiClient = new ApiClient('.');

export { t, config, apiClient };
