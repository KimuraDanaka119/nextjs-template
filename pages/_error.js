import React from 'react';
import PropTypes from 'prop-types';
import Error from 'next/error';

import Layout from '../components/Layout';
import Meta from '../components/Meta';
import withI18N from '../components/withI18N';

class MyError extends Error {
  static getInitialProps({ res, err }) {
    const { statusCode } = res || err;

    return { statusCode };
  }

  render() {
    const { t, statusCode } = this.props;

    return (
      <Layout>
        <Meta
          title={t('error:meta.title')}
          description={t('error:meta.description')}
        />
        <div className="error">
          <p>
            {this.props.statusCode
              ? t('error:content.withStatusCode', { statusCode })
              : t('error:content.withoutStatusCode')
            }
          </p>
        </div>
      </Layout>
    );
  }
}

MyError.propTypes = {
  t: PropTypes.func,
  statusCode: PropTypes.number,
};

export default withI18N(MyError, ['error']);
