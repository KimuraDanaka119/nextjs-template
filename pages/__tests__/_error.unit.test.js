import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

import { undecorated as ErrorPage } from '../_error';
import { t } from '../../lib/testHelper';

describe('<ErrorPage />', () => {
  it('getInitialProps should return object with status code', () => {
    let props = ErrorPage.getInitialProps({ res: { statusCode: 404 } });
    expect(props).toEqual({ statusCode: 404 });

    props = ErrorPage.getInitialProps({ err: { statusCode: 404 } });
    expect(props).toEqual({ statusCode: 404 });

    props = ErrorPage.getInitialProps({
      res: { statusCode: 404 },
      err: { statusCode: 500 },
    });

    expect(props).toEqual({ statusCode: 404 });
  });

  it('should render correctly without status code', () => {
    const error = shallow(<ErrorPage t={t} />);
    expect(toJson(error)).toMatchSnapshot();
  });

  it('should render correctly with status code', () => {
    const error = shallow(<ErrorPage t={t} statusCode={404} />);
    expect(toJson(error)).toMatchSnapshot();
  });
});
