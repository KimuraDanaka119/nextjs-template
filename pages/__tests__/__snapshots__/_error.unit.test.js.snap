// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error /> should render correctly with status code 1`] = `
Array [
  <div
    class="header"
  >
    <div
      class="container"
    >
      <a
        href="/"
      >
        <img
          src="/static/47cd76e43f74bbc2e1baaf194d07e1fa/images/favicon.png"
        />
      </a>
      <span>
        Batman TV Programs
      </span>
    </div>
  </div>,
  <div
    class="container"
  >
    <p>
      An error 404 occurred on server
    </p>
  </div>,
  <div
    class="footer"
  >
    <div
      class="container"
    >
      This is footer here
    </div>
  </div>,
]
`;

exports[`<Error /> should render correctly without status code 1`] = `
Array [
  <div
    class="header"
  >
    <div
      class="container"
    >
      <a
        href="/"
      >
        <img
          src="/static/47cd76e43f74bbc2e1baaf194d07e1fa/images/favicon.png"
        />
      </a>
      <span>
        Batman TV Programs
      </span>
    </div>
  </div>,
  <div
    class="container"
  >
    <p>
      An error occurred on client
    </p>
  </div>,
  <div
    class="footer"
  >
    <div
      class="container"
    >
      This is footer here
    </div>
  </div>,
]
`;
